### ISSUE TITLE
Legend items not navigable with screen readers

### ISSUE THREAD

**Issue by oysteinmoseng** [highcharts/highcharts#6512]
Using the a11y module, the legend items should be exposed to screen readers, with the appropriate AT markup (`role="button"` and `aria-pressed=true/false`).

If this is implemented, we can consider removing the `aria-hidden` and even the label of the parent element.

**Comment by JamesCatt** [comment]
Can I ask what the status of this is? It looks like legends have the role set in the current version (7.0.1), but the keyboard interaction doesn't work when using a screen reader, despite being announced as a button. Works as expected when screen reader isn't running though.

**Comment by oysteinmoseng** [comment]
@JamesCatt We are aiming to implement this by the end of the week. Keyboard interaction with the screen reader running can be a bit tricky, since the screen reader captures all of the key events, but we should be able to make it work.

**Comment by oysteinmoseng** [comment]
This is now implemented, but not yet working in Narrator. I am closing this, and we can follow up in #9414. See that issue for a demo of the current implementation.

**Comment by JamesCatt** [comment]
If you set role="application" instead of region on the chart container, that will stop the screen reader from intercepting the key events and allows the chart interaction to function properly.

**Comment by oysteinmoseng** [comment]
@JamesCatt It would, but then the users would be unable to navigate the chart using their default screen reader shortcuts, which would be confusing. We are trying to avoid using the application role so that users do not have to learn a new set of navigation keys to navigate the charts.

**Comment by JamesCatt** [comment]
@oysteinmoseng I see. To clarify though, does your upcoming fix also fix the arrow key interactions as well? The screen reader instructions say to use arrow keys but currently they don't work in either JAWS or NVDA.

I don't see an open issue about it, but I didn't want to open a new issue if it's already being addressed in the upcoming release.

**Comment by oysteinmoseng** [comment]
@JamesCatt The arrow key hint is something we are considering removing, as screen reader users will already know the navigation shortcuts for the screen reader they are using. The hint refers specifically to using "down arrow" to navigate to the next element for JAWS and NVDA.

Using left/right arrow keys to move between the points will not work when a screen reader is running, unless we decide to implement that with a role `application` approach.
